pkgs/desktops/gnome-3/extensions/update-extensions.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+import urllib.request, urllib.error, json, types
+from typing import List, Set, Dict, Optional
+
+# We don't want all those deprecated legacy extensions
+# Group extensions by Gnome version for compatibility reasons
+supported_versions = {
+    "36": {
+        "3.36",
+        "3.36.0",
+        "3.36.1",
+        "3.36.2",
+        "3.36.3",
+        "3.36.4",
+        "3.36.5",
+        "3.36.6",
+    },
+    "38": {"3.38", "3.38.0", "3.38.1"},
+}
+
+
+def generate_extension_versions(
+    extension_version_map: Dict[str, str], uuid: str
+) -> Dict[str, Dict[str, str]]:
+    extension_versions: Dict[str, Dict[str, str]] = {}
+    for shell_version, sub_versions in supported_versions.items():
+        # Newest compatible extension version
+        extension_version: Optional[int] = max(
+            (
+                int(ext_ver)
+                for shell_ver, ext_ver in extension_version_map.items()
+                if (shell_ver in sub_versions)
+            ),
+            default=None,
+        )
+        # Extension is not compatible with this Gnome version
+        if not extension_version:
+            continue
+        print(f"[{shell_version}] Downloading {uuid}, {str(extension_version)}")
+        sha256 = fetch_sha256sum(uuid, str(extension_version))
+        extension_versions[shell_version] = {
+            "version": str(extension_version),
+            "sha256": sha256,
+        }
+    return extension_versions
+
+
+# Download the extension and hash it
```

Comment:
Just make this a proper Python docstring if this comment is supposed to document the purpose of the function.